cypress/full/typeahead_page_spec.ts

```diff
@@ -0,0 +1,369 @@
+    const textToInput = 'gia ge';
+    const toggleBtnText = 'Toggle typeaheadSingleWords';
+
+    it('a user sees the btn "Toggle typeaheadSingleWords" and the card with default state "typeaheadSingleWords: true"',
+      () => {
+        typeahead.scrollToMenu('Ignore spaces and order');
+        typeahead.isElementVisible(ignoreSpace, typeahead.cardHeader);
+        typeahead.isPreviewExist(ignoreSpace, formTemplate);
+        typeahead.isElementVisible(ignoreSpace, typeahead.btnSelector);
+        typeahead.isElemTextCorrect(ignoreSpace, typeahead.btnSelector, toggleBtnText);
+        typeahead.isElementVisible(ignoreSpace, typeahead.inputSelector);
+      });
+
+    it('when a user types then order of typed symbols and spaces between them is ignored', () => {
+      typeahead.clearInputAndSendKeys(ignoreSpace, textToInput);
+      typeahead.isDropdownEnabled(typeahead.activeDropdown, 'be.visible');
+      typeahead.isElemTextCorrect(ignoreSpace, typeahead.dropdownBtn, stateForCheck);
+
+      typeahead.isSelectedItemDisplayed(ignoreSpace, stateForCheck, formTemplateFilled);
+    });
+
+    it('a user toggles symbols and spaces ignoring off', () => {
+      typeahead.clickOnBtn(ignoreSpace);
+      typeahead.isPreviewExist(ignoreSpace, formTemplateFalse);
+      typeahead.clearInputAndSendKeys(ignoreSpace, textToInput);
+      typeahead.isDropdownEnabled(typeahead.activeDropdown, 'not.be.enabled');
+    });
+  });
+
+  describe('Phrase delimiters', () => {
+    const phraseDelimeters = typeahead.exampleDemosArr.ignoreSpaceAndOrder;
+    const stateForCheck = 'New York';
+    const textToInput = 'york new';
+    const textWithDelimiters = '&york new&';
+
+    it('a user is able to use "&" and "," to delimit words for exact match',
+      () => {
+        typeahead.scrollToMenu('Phrase delimiters');
+        typeahead.isElementVisible(phraseDelimeters, typeahead.inputSelector);
+        typeahead.clearInputAndSendKeys(phraseDelimeters, textWithDelimiters);
+        typeahead.isDropdownEnabled(typeahead.activeDropdown, 'not.be.enabled');
+      });
+
+    it('using the same phrase without delimiters', () => {
+      typeahead.clearInputAndSendKeys(phraseDelimeters, textToInput);
+      typeahead.isDropdownEnabled(typeahead.activeDropdown, 'be.visible');
+
+      typeahead.isSelectedItemDisplayed(phraseDelimeters, stateForCheck, stateForCheck);
+    });
+  });
+
+  describe('Dropup', () => {
+    const dropUp = typeahead.exampleDemosArr.dropUp;
+    const stateForCheck = 'Mississippi';
+    const textToInput = 'p';
+
+    it('the list (drop-up) with matches is shown above the input',
+      () => {
+        typeahead.scrollToMenu('Dropup');
+        typeahead.isElementVisible(dropUp, typeahead.inputSelector);
+        typeahead.clearInputAndSendKeys(dropUp, textToInput);
+        typeahead.isDropdownEnabled(typeahead.activeDropdown, 'be.visible');
+        // TODO need to test that the dropup appears above the input
+
```

Comment:
as example for implementation use method isDatepickerPlacementCorrect from datepicker